tensorboard/webapp/metrics/views/main_view/main_view_test.ts

```diff
@@ -894,6 +895,75 @@ describe('metrics main view', () => {
         ).toBe(0);
       });
     });
+
+    describe('card width setting', () => {
+      beforeEach(() => {
+        store.overrideSelector(selectors.getNonEmptyCardIdsWithMetadata, [
+          {
+            cardId: 'card1',
+            plugin: PluginType.SCALARS,
+            tag: 'tagA',
+            runId: null,
+          },
+        ]);
+      });
+
+      it('does not set the max width without feature flag enabled', () => {
+        const fixture = TestBed.createComponent(MainViewContainer);
+        fixture.detectChanges();
+
+        expect(
+          fixture.debugElement.query(By.css('.card-grid')).styles[
+            'grid-template-columns'
+          ]
+        ).toBe('');
+      });
+
+      it('sets the max width to be cardMaxWidth', () => {
+        store.overrideSelector(selectors.getEnabledCardWidthSetting, true);
+        store.overrideSelector(selectors.getMetricsCardMaxWidth, 50);
+        const fixture = TestBed.createComponent(MainViewContainer);
+        fixture.detectChanges();
+
+        expect(
+          fixture.debugElement.query(By.css('.card-grid')).styles[
+            'grid-template-columns'
+          ]
+        ).toBe('repeat(auto-fill, minmax(335px, 50vw))');
+      });
+
+      it('does not set the max width with invalid width value', () => {
```

Comment:
Just for completeness, and given that it will be the default behavior, would you mind adding a test where getEnabledCardWidthSetting is true and getMetricsCardWidth is null?